src/analyses/base.ml

```diff
@@ -2328,6 +2353,28 @@ struct
           | _ -> failwith ("non-floating-point argument in call to function "^f.vname)
         end
       in
+      let apply_abs ik x =
+        let eval_x = eval_rv (Analyses.ask_of_ctx ctx) gs st x in
+        begin match eval_x with
+          | Int int_x ->
+            let xcast = ID.cast_to ik int_x in
+            (* the absolute value of the most-negative value is out of range for 2'complement types *)
+            (match (ID.minimal xcast), (ID.minimal (ID.top_of ik)) with
+             | _, None
+             | None, _ -> ID.top_of ik
+             | Some mx, Some mm when Z.equal mx mm -> ID.top_of ik
```

Comment:
I think this should also check the signedness because for unsigned types the minimal value is `0`, which is completely fine.

Also, this seems a bit roundabout way to do `ID.equal_to`. That would be better because it could benefit from the non-interval domain as well. For example, if it cannot be `INT_MIN` by congruence.

Reply:
Can the type ever be unsigned here? I thought it is cast to `int` by CIL?

Reply:
Actually, this function is currently only called with `IInt` anyway.

Reply:
Now with @stilscher's changes the function may be called with `ILong` and `ILongLong` as well.

Reply:
`ILong` and `ILongLong` should be signed as well, right? So that should not make a difference.

Reply:
Yes, I think one then does not need to handle the case of having an unsigned type, if all variations of `abs` functions take signed types only.